quickwit/quickwit-indexing/src/source/queue_sources/mod.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+pub mod coordinator;
+mod helpers;
+mod local_state;
+#[cfg(test)]
+mod memory_queue;
+mod message;
+mod shared_state;
+#[cfg(feature = "sqs")]
+pub mod sqs_queue;
+#[cfg(feature = "sqs")]
+pub mod sqs_source;
+mod visibility;
+
+use std::fmt;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use async_trait::async_trait;
+use message::RawMessage;
+
+type ReceiveResult = anyhow::Result<Vec<RawMessage>>;
```

Comment:
I suspect this typedef makes the code harder to read.

Reply:
Yes, I'm also sceptical about this. I needed something because the linter was complaining that `Option<BoxFuture<'static, ReceiveResult>>anyhow::Result<Vec<RawMessage>>` was too long. I'll go for a more local typedef.